### PR TITLE
New Tech Tree

### PR DESCRIPTION
**Pull request by MTandi**
## About The Pull Request

Reshuffles the tech tree nodes, adding reagent purity and cybernetic organ scan experiments. The total point requirements barely changed.

Made the discount experiments unlock nodes for free, instead of providing partial discounts, to provide more incentive to actually do them.

Also devided all points by the amount generated per second, for convenience. Now points correspond to seconds.

<details> 
  <summary>New tech tree</summary>
Tech New
</details>

## Why It's Good For The Game

- Breaks bloated general nodes into more specialized nodes, making the tree more balanced
- Combines scattered nodes dedicated to a single design into specialized nodes, reducing the number of nodes
- Reshuffles the unlocks within specialized tree to adjust progression on some trees to follow the idead: Cheap nodes first, then nodes that require experiments or nodes of neighbouring trees, then expensive nodes with end-game tech.
- You no longer need a dissection experiment to unlock a microwave, every experiment is relevant to the tree unlocked by it
- With specialized nodes, it is easier to pick a node for the new things that people add
- Better foundation for potential per-department point system. It will be easier to put it on top of a cleaner tree
- The nodes are separated in tiers, with progressively increasing costs, so that early tech is cheap and easy to research with just points, but later tech is more expensive, making the discount experiments for those more likely to be performed.

## Changelog

:cl:
balance: Reshuffled tech tree, making nodes more specialized
qol: Research points devided by the amount generated per second, so now research points correspond to seconds
add: Introduced reagent purity scan experiments (required for Cryostasis node)
add: Introduced synthetic organ scan experiment (required for top tier cyber organs)
add: Added a variant of machinery scan experiment that accepts any machines with upgraded parts (required for tier 3 parts)
del: Removed material scanning experiments from the tech tree
/:cl:

### CODE REVIEW

**Comment by MTandi** [comment]
Drafted for feedback.
Note the discord dev project discussion: https://discord.com/channels/326822144233439242/1248189914449973258

**Comment by DATA-xPUNGED** [comment]
image
I think you copied the wrong thing here

**Comment by MTandi** [comment]
> image I think you copied the wrong thing here

The image is just for referrence, I'm lazy to paint every single detail into the chart. These nodes are the same as in the old tree.

File: code/modules/research/techweb/all_nodes.dm
Comment (ZephyrTFA) [inline code/modules/research/techweb/all_nodes.dm:1]: alphabetize all of the lists.
no comment headers
split up into subfiles

**Review by ZephyrTFA — CHANGES_REQUESTED** [review]

**Comment by zoomachina** [comment]
Maybe the plasma cutter shouldn't be hidden behind 6 nodes

**Comment by Vect0r2** [comment]
I don't like how genetics is behind two other experiments that they have no control over.

**Comment by MTandi** [comment]
> Maybe the plasma cutter shouldn't be hidden behind 6 nodes

It is about the same as before in terms of points due to industrial engineering needing 12 500 points

**Comment by zoomachina** [comment]
> > Maybe the plasma cutter shouldn't be hidden behind 6 nodes
> 
> It is about the same as before in terms of points due to industrial engineering needing 12 500 points

Industrial engineering has the easiest experiment with a 10k discount, it's practically 2.5k if scientists have the smallest amount of experience. This won't be the same here, even if all of these nodes can be paid for with experiments

File: code/modules/research/techweb/all_nodes.dm [hunk 1]
```diff
@@ -1,1075 +1,647 @@
...
+		"earmuffs",
+		"recorder",
+		"tape",
+		"toy_balloon",
+		"pet_carrier",
+		"chisel",
+		"spraycan",
+		"camera_film",
+		"camera",
+		"razor",
```
[the remark this answers is not in the document]
Comment (MTandi) [reply code/modules/research/techweb/all_nodes.dm:39]: They're ordered manually to appear grouped in UI, so this is on purpose.

**Comment by MTandi** [comment]
> I don't like how genetics is behind two other experiments that they have no control over.

You can do both of them in sci:

image

You can make a slime by taking biopsy from mapspawn slimes, and grow one with protein from tofu.
And the plant can be grown on this patch. There is a maint garden nearby with seeds and left4z, or you can mutate a random weed that spawns on this patch.

Besides, geneticists already have everything needed roundstart, so they don't need this node early anyway.

**Comment by MTandi** [comment]
> Industrial engineering has the easiest experiment with a 10k discount, it's practically 2.5k if scientists have the smallest amount of experience. This won't be the same here, even if all of these nodes can be paid for with experiments

Will probably move it to Controlled Plasma to avoid backlash, or introduce discounts on the way. But personally I believe it is in the right place in terms of balance.

File: code/modules/research/techweb/all_nodes.dm [hunk 2]
```diff
@@ -404,7 +404,7 @@
 		"splitbeaker",
 	)
 	research_costs = list(TECHWEB_POINT_TYPE_GENERIC = TIER_3_POINTS)
-	required_experiments = list(/datum/experiment/scanning/reagent/cryoxadone)
+	required_experiments = list(/datum/experiment/scanning/reagent/cryostaline)
```
Comment (zoomachina) [inline code/modules/research/techweb/all_nodes.dm:407]: cryostaline -> cryostylane

**Comment by githubuser4141** [comment]
Shouldn't the basic mech firearms be researched later, not sooner? The ULTRA-AC2 and Scattershot are incredibly deadly and have the highest DPS in most situations, the AC2 being the best general mech weapon (60 damage a burst, 1200 bullets, 1 second delay). Mech lasers/ions are underwhelming, the standard mech laser fires a single laser at a slow rate of fire, the cannon (40 damage) actually has less DPS due to being twice as slow to fire as the basic laser (25 damage).

The 'heavy' mech weapons are actually fairly worse in combat. Rockets only explode if they strike a solid object like a mech or a wall, otherwise they have 6 shots with no spare magazine, have a long cooldown and only do 20 damage to a person. The flashbang/clusterbang is occasionally useful but still suffers from lack of ammo and flashbangs that seem to blind a person in a mech.

**Comment by CRITAWAKETS** [comment]
Some of these are a little strangely placed in the higher tiers, like why are Bitrunning, regular RCD upgrades and Engi/Mining Cyborg upgrades all going to cost 10000 points? That's a bit nuts. Especially the engi/mining borg upgrades, given how mining borgs shine more in the early game.

**Comment by atlasle** [comment]
can you put that tech tree visual in the game

**Comment by MTandi** [comment]
> can you put that tech tree visual in the game

I've built the tree keeping in mind that we can visualize it with nodes that we have for circuits later. That's why some techs have forced tier - to make sure that all required nodes are on a lower or the same tier.

But I'm not the best to code this into the game as I'm not familiar with node component.

**Comment by Exester509** [comment]
I feel like some researches are too late into the tech tree, namely plasma cutters and engineering cyborg upgrades.
MODsuit nodes are also weirdly renamed into exosuits, which used to be the mechs, but that isn't really a concern.

Finally, why is illegal tech gatekept _behind_ unregulated bluespace research and not the other way around? The desynchronizer is one of the most powerful items science can produce, being able to accomplish true inmortality. There's a reason why it's so rare

**Comment by moocowswag** [comment]
Im okay with where shield gens were put

**Comment by MTandi** [comment]
> I feel like some researches are too late into the tech tree, namely plasma cutters and engineering cyborg upgrades. 
Will be moved closer

>MODsuit nodes are also weirdly renamed into exosuits, which used to be the mechs, but that isn't really a concern.
There is a confusion between modsuits and exosuits, as in scifi, piloted robots are more commonly called "Mech" and anything with "-suit" is about armored exoskeletons. Took the liberty to call all piloted robots "Mech" and MOD suits as "Exosuit". The other in-game machines are also inconsistent in this naming.

> Finally, why is illegal tech gatekept _behind_ unregulated bluespace research and not the other way around? The desynchronizer is one of the most powerful items science can produce, being able to accomplish true inmortality. There's a reason why it's so rare
Error in the chart graphics, it's the same as before.

**Comment by MTandi** [comment]
> 10k points for polybelt and all the prerequisites are going to kill it though.

Will merge into the Advanced Anomaly Shells and plan to introduce an anomaly scan discount experiment.

> T4 parts coming earlier in the tree is probably going to be the biggest impact.

Considering updating machine scanning experiments to accept ANY machine with the previous tier parts to unlock the bluespace part.

> Oh wow wait exotic ammunition is also no longer soft removed??? this means tech shells are probably going to see consistent usage in rev cult and nukie rounds.

Considering an experiment to unlock them, requiring side research, or moving to the later part of the tree to cost more

> Engineering and mining borg upgrades becoming a 10k point research thats far down the tree with many prerequisites ... these are tools that are meant to use longterm, you dont want to wait till the round is over to suddenly get the ability to mine better or build machines.

Moving closer.

**Comment by norsvenska** [comment]
what if the chromatography machine was the thing that did the purity experiments? it makes more sense imo than the regular experiscanner

**Comment by MTandi** [comment]
> what if the chromatography machine was the thing that did the purity experiments? it makes more sense imo than the regular experiscanner plus, it would give it another use. i haven't seen many people use it.

Good idea. Please code it to do so.

**Comment by MTandi** [comment]
Updated the tree and introduced some new experiments.

**Comment by CRITAWAKETS** [comment]
Advanced plasma cutters should probably not be in the same tier as the regular plasma cutter. Honestly putting it in advanced mining with a discount experiment would likely work great, especially since advanced mining currently has the issue of only being attractive if you know a secret gamer technique and nothing else.

**Comment by moocowswag** [comment]
Bcis should be another separate branch alongside drone shells and programmed servers rather than being gated behind drone shells, research is not meant to be what gatekeeps circuits as they themselves are their own subdepartment that requires alot of undivided attention to work on, comparable to genetics and viro. There is no "passively wait on progress" you have to invest a large portion of the round into making them and get them early.

Or just add discount experiments to them related to making an easy circuit

I can already see myself getting lynched because this pr makes me spend ALOT more points to get bcis which only i'll be using, currently I spend 3k research points TOTAL throughout an entire round on circuits because im able to just grab the starting tech and bci and i dont touch drones or server shell and i still get yelled at.

**Comment by MTandi** [comment]
> Advanced plasma cutters should probably not be in the same tier as the regular plasma cutter. Honestly putting it in advanced mining with a discount experiment would likely work great, especially since advanced mining currently has the issue of only being attractive if you know a secret gamer technique and nothing else.

Advanced one is on tier 3 together with the mech variant. Regular is on tier 2 now. 
Advanced mining is appealing at least in the diamond drill that breaks reinforced walls. I don't know the secret gamer technique, but I think that diamond drills and jackhammer should be buffed or given a secondary utility, as they're not nearly as good as plasma cutters for mining.

**Comment by MTandi** [comment]
> TLDR: Just remove drone shells/programmed bot as a prerequisite for bci and maybe add a discount experiment.

image

**Comment by moocowswag** [comment]
So usually scan both roboticists/the rd/engies/alot of jobs really or use the entertainment vendor to get the toy skillchips

**Comment by Maximal08** [comment]
I don't really like the option with the bluespace rped only when studying 4 tiers of parts, because the usual rped is not the most convenient thing, and such a late study of a more convenient alternative will make the life of those who assemble machinery more difficult.

**Comment by MTandi** [comment]
> So usually scan options are both roboticists/the rd/engies/alot of jobs really or use the entertainment vendor to get the toy skillchips

Any skillchips, so you can just scan a clown and a roboticist.

**Comment by MTandi** [comment]
> I don't really like the option with the bluespace rped only when studying 4 tiers of parts, because the usual rped is not the most convenient thing, and such a late study of a more convenient alternative will make the life of those who assemble machinery more difficult. but on the other hand, 4 thousand details appear earlier.

It's a purpose of tech tree to make people use less convenient things, do research, unlock more convenient things one by one and be happy that what they did just became more convenient with the research. 

That's why I'm a fan of moving plasma cutters to the later nodes. But I don't want to be that guy that made miners suffer more than before this PR. Let maintainers decide this kind of balance things.

**Comment by Maximal08** [comment]
> > I don't really like the option with the bluespace rped only when studying 4 tiers of parts, because the usual rped is not the most convenient thing, and such a late study of a more convenient alternative will make the life of those who assemble machinery more difficult. but on the other hand, 4 thousand details appear earlier.
> 
> It's a purpose of tech tree to make people use less convenient things, do research, unlock more convenient things one by one and be happy that what they did just became more convenient with the research.
> 
> That's why I'm a fan of moving plasma cutters to the later nodes. But I don't want to be that guy that made miners suffer more than before this PR. Let maintainers decide this kind of balance things.
> 
> You don't usually need to upgrade more than a couple of machines for your job, so you won't die if you have to use a screwdriver a couple of times. And parts provide linear efficiency boost, rather than an instant milestone like plasma cutters. So it is fine to not have the t4 with bluespace RPED unlocked ASAP.

I generally like these changes, although advanced and conventional cutters in the same technology raise questions, and I'm looking forward to it in the game with the goal of seeing what happens.

**Comment by MTandi** [comment]
> I generally like these changes, although advanced and conventional cutters in the same technology raise questions, and I'm looking forward to it in the game with the goal of seeing what happens.

The regular one is T2 (Controlled plasma node), the upgraded one is T3 (Plasma beam mining). The one next to it is a mech variant.

**Comment by Maximal08** [comment]
> > I generally like these changes, although advanced and conventional cutters in the same technology raise questions, and I'm looking forward to it in the game with the goal of seeing what happens.
> 
> The regular one is T2 (Controlled plasma node), the upgraded one is T3 (Plasma beam mining). The one next to it is a mech variant. Wrong icon in the tree.

a funny mistake, which is why this change seemed strange to me.

**Comment by ZephyrTFA** [comment]
fuck it

**Comment by LSommers** [comment]
This PR is great if only because I can laugh at people salting in OOC that Research and Development has to Research and Develop now.
